### PR TITLE
[473374] Fixed NPE

### PR DESCRIPTION
**Pull request by svenefftinge**

### CODE REVIEW

**Comment by eclipsewebmaster** [comment]
Issue tracker reference:
https://bugs.eclipse.org/bugs/show_bug.cgi?id=473374

**Comment by svenefftinge** [comment]
Jenkins intellij-tests-master-latest

File: intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/highlighting/SemanticHighlightVisitor.xtend [hunk 1]
```diff
@@ -73,8 +70,9 @@ class SemanticHighlightVisitor implements HighlightVisitor {
 	}
 	
 	override clone() {
-		cloneProvider.get => [
+		getClass.newInstance => [
```
Comment (szarnekow) [inline intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/highlighting/SemanticHighlightVisitor.xtend]: Implementing Cloneable and calling super.clone should make sure that all fields are copied regardless of specializations in subtypes. I would expect the clone doesn't have a languageID od highlightCalculator right now

Comment (szarnekow) [inline intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/highlighting/SemanticHighlightVisitor.xtend]: My bad - subtypes are supposed to have a constructor with injectMembers(this), right?

Comment (svenefftinge) [inline intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/highlighting/SemanticHighlightVisitor.xtend]: yes

**Pull request merged**